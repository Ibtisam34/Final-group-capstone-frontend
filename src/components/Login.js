import { useRef } from 'react';
import PropTypes from 'prop-types';

const Login = ({ setCurrUser, setShow }) => {
  const formRef = useRef();
  const login = async (userInfo, setCurrUser) => {
    const url = 'http://localhost:3000/users/sign_in';
    try {
      const response = await fetch(url, {
        method: 'post',
        headers: {
          'content-type': 'application/json',
          accept: 'application/json',
        },
        body: JSON.stringify(userInfo),
      });
      const data = await response.json();
      if (!response.ok) throw data.error;
      localStorage.setItem('token', response.headers.get('Authorization'));
      setCurrUser(data);
    } catch (error) {
      console.log('error', error);
    }
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    const formData = new FormData(formRef.current);
    const data = Object.fromEntries(formData);
    const userInfo = {
      user: { email: data.email, password: data.password },
    };
    login(userInfo, setCurrUser);
    e.target.reset();
  };
  const handleClick = (e) => {
    e.preventDefault();
    setShow(false);
  };
  return (
    <div>
      <form ref={formRef} onSubmit={handleSubmit}>
        Email:
        {' '}
        <input type="email" name="email" placeholder="email" />
        <br />
        Password:
        {' '}
        <input type="password" name="password" placeholder="password" />
        <br />
        <input type="submit" value="Login" />
      </form>
      <br />
      <div>
        Not registered yet,
        <a href="#signup" onClick={handleClick}>Signup</a>
      </div>
    </div>
  );
};
Login.propTypes = {
  setCurrUser: PropTypes.string,
  setShow: PropTypes.shape([]).isRequired,
};
Login.defaultProps = {
  setCurrUser: '',
};
export default Login;
