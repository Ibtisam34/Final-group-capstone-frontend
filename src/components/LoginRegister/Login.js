import React, { useEffect, useState } from 'react';
import { useNavigate } from 'react-router-dom';
import Container from 'react-bootstrap/Container';
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';
import { useSelector, useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import Slider from './Slider';
import Logo from '../../assets/images/LOGO-DOCTOR.png';
import { fetchUsers } from '../../redux/users/usersReducer';

function Login({ setIsLoggedIn }) {
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const [userEmail, setEmail] = useState('');

  useEffect(() => {
    dispatch(fetchUsers());
  }, [dispatch]);
  const users = useSelector((state) => state.usersReducer);

  const handleLogin = () => {
    users.map((user) => {
      if (user.email === userEmail) {
        localStorage.setItem('isLoggedIn', 'true');
        localStorage.setItem('userId', `${user.id}`);
        setIsLoggedIn('true');
        navigate('/');
      }
      return 'hey';
    });
  };

  return (
    <Container fluid>
      <Container className="login-card col-4 pb-5 bg-white border rounded">
        <div className="logo-wrapper">
          <img src={Logo} alt="logo" className="logo-img" width="200" height="200" />
        </div>
        <Form>
          <Form.Group className="my-3" controlId="formBasicEmail">
            <Form.Control type="email" placeholder="Email address" required onChange={(e) => setEmail(e.target.value)} />
          </Form.Group>

          <Form.Group className="mb-3 text-center">
            <Button variant="success" type="submit" onClick={() => handleLogin()}>
              Login
            </Button>
          </Form.Group>
        </Form>
        <div>
          <p>
            Don&apos;t have an account?
            {' '}
            <span>
              <button
                type="button"
                onClick={() => { navigate('/register'); }}
                className="access-btn text-primary"
              >
                Register
              </button>
            </span>
          </p>
        </div>
      </Container>
      <Slider />
    </Container>
  );
}

Login.propTypes = {
  setIsLoggedIn: PropTypes.func.isRequired,
};

export default Login;
