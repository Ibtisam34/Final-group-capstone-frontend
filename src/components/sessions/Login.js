import { useRef } from 'react';
import PropTypes from 'prop-types';
import { login } from './api';

const Login = ({ setCurrUser, setShow }) => {
  const formRef = useRef();

  const handleSubmit = async (e) => {
    e.preventDefault();
    const formData = new FormData(formRef.current);
    const data = Object.fromEntries(formData);
    const userInfo = {
      user: { email: data.email, password: data.password },
    };
    await login(userInfo, setCurrUser);
    e.target.reset();
  };
  const handleClick = (e) => {
    e.preventDefault();
    setShow(false);
  };
  return (
    <div>
      <form ref={formRef} onSubmit={handleSubmit}>
        Email:
        {' '}
        <input type="email" name="email" placeholder="email" />
        <br />
        Password:
        {' '}
        <input type="password" name="password" placeholder="password" />
        <br />
        <input type="submit" value="Login" />
      </form>
      <br />
      <div>
        Not registered yet,
        <a href="#signup" onClick={handleClick}>Signup</a>
      </div>
    </div>
  );
};
Login.propTypes = {
  setCurrUser: PropTypes.func.isRequired,
  setShow: PropTypes.func.isRequired,
};
export default Login;
