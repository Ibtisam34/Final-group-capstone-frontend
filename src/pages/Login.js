import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { signIn, loggedin } from '../redux/reducer/user/userSlice';
import 'bootstrap/dist/css/bootstrap.min.css';

const Login = () => {
  const [user, setUser] = useState({});

  const dispatch = useDispatch();
  const navigate = useNavigate();
  const isLoggedIn = useSelector(loggedin);

  const handleChange = (e) => {
    const {
      target: { name: input, value },
    } = e;
    setUser({ ...user, [input]: value });
  };

  const handleSignIn = async (e) => {
    e.preventDefault();
    const authLogin = await dispatch(signIn(user));
    if (authLogin.payload.status === 'successful') {
      localStorage.setItem('authUser', JSON.stringify(authLogin.payload.user));
      navigate('/');
    }
  };

  useEffect(() => {
    if (isLoggedIn) navigate('/');
  }, [isLoggedIn]);

  document.title = 'Famous Doctors | Login';

  return (
    <div className="container-fluid mt-5">
      <div className="row justify-content-center">
        <div className="col-md-6">
          <div className="card">
            <div className="card-header text-center">
              <h3>Login</h3>
            </div>
            <div className="card-body">
              <form onSubmit={handleSignIn}>
                <div className="form-group mb-3">

                  <label htmlFor="email" style={{ width: '100%' }}>
                    Email address
                    <input
                      name="email"
                      type="email"
                      className="form-control"
                      id="email"
                      onChange={handleChange}
                    />

                  </label>
                </div>
                <div className="form-group mb-3">
                  <label htmlFor="password" className="form-label" style={{ width: '100%' }}>
                    Password
                    <input
                      type="password"
                      name="password"
                      className="form-control"
                      id="passwordInput"
                      onChange={handleChange}
                    />
                  </label>
                </div>
                <button type="button" className="btn btn-primary" onClick={handleSignIn}>Login </button>
              </form>
              <div className="mt-3">
                Don&apos;t have an account?
                {' '}
                <a href="/register">Register</a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Login;
