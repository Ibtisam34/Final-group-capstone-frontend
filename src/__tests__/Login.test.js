import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import Login from '../components/LoginRegister/Login'


describe('Login to app', () => {
  it('renders correctly', () => {
    const tree = render(<Provider store={store}><Router><Login /></Router></Provider>);
    expect(tree).toMatchSnapshot();
  });

  
  it('should have email input field', () => {
    const { getByPlaceholderText } = render(<Provider store={store}><Router><Login /></Router></Provider>);
    const emailInput = getByPlaceholderText('Email address');
    expect(emailInput).toBeInTheDocument();
  })


});