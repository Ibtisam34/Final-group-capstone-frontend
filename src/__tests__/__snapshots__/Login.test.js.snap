// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login to app renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container-fluid"
      >
        <div
          class="login-card col-4 pb-5 bg-white border rounded container"
        >
          <div
            class="logo-wrapper"
          >
            <img
              alt="logo"
              class="logo-img"
              height="200"
              src="LOGO-DOCTOR.png"
              width="200"
            />
          </div>
          <form
            class=""
          >
            <div
              class="my-3"
            >
              <input
                class="form-control"
                id="formBasicEmail"
                placeholder="Email address"
                required=""
                type="email"
              />
            </div>
            <div
              class="mb-3 text-center"
            >
              <button
                class="btn btn-success"
                type="submit"
              >
                Login
              </button>
            </div>
          </form>
          <div>
            <p>
              Don't have an account?
               
              <span>
                <button
                  class="access-btn text-primary"
                  type="button"
                >
                  Register
                </button>
              </span>
            </p>
          </div>
        </div>
        <div
          class="carousel carousel slide"
        >
          <div
            class="carousel-indicators"
          >
            <button
              aria-current="true"
              aria-label="Slide 1"
              class="active"
              data-bs-target=""
              type="button"
            />
            <button
              aria-current="false"
              aria-label="Slide 2"
              data-bs-target=""
              type="button"
            />
            <button
              aria-current="false"
              aria-label="Slide 3"
              data-bs-target=""
              type="button"
            />
            <button
              aria-current="false"
              aria-label="Slide 4"
              data-bs-target=""
              type="button"
            />
            <button
              aria-current="false"
              aria-label="Slide 5"
              data-bs-target=""
              type="button"
            />
            <button
              aria-current="false"
              aria-label="Slide 6"
              data-bs-target=""
              type="button"
            />
            <button
              aria-current="false"
              aria-label="Slide 7"
              data-bs-target=""
              type="button"
            />
          </div>
          <div
            class="carousel-inner"
          >
            <div
              class="carousel-item active carousel-item"
            >
              <img
                alt="travel vacation"
                class="d-block w-100"
                src="https://images.unsplash.com/photo-1629909613654-28e377c37b09?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxzZWFyY2h8Mnx8Y2xpbmljfGVufDB8fDB8fA%3D%3D&w=1000&q=80"
              />
            </div>
            <div
              class="carousel-item carousel-item"
            >
              <img
                alt="travel vacation"
                class="d-block w-100"
                src="https://e-clinic.co.uk/wordpress_new/wp-content/uploads/2021/09/Image-14-09-2021-at-13.59.jpg"
              />
            </div>
            <div
              class="carousel-item carousel-item"
            >
              <img
                alt="travel vacation"
                class="d-block w-100"
                src="https://firstmedcenters.com/wp-content/uploads/2022/07/English-speaking-clinic-FirstMed-V2-2.jpg"
              />
            </div>
            <div
              class="carousel-item carousel-item"
            >
              <img
                alt="travel vacation"
                class="d-block w-100"
                src="https://www.extraspace.com/blog/wp-content/uploads/2020/10/preparing-home-for-airbnb.jpg"
              />
            </div>
            <div
              class="carousel-item carousel-item"
            >
              <img
                alt="travel vacation"
                class="d-block w-100"
                src="https://img1.wsimg.com/isteam/stock/4810/:/cr=t:0%25,l:5.83%25,w:88.34%25,h:100%25"
              />
            </div>
            <div
              class="carousel-item carousel-item"
            >
              <img
                alt="travel vacation"
                class="d-block w-100"
                src="https://healthwaymedical.com/wp-content/uploads/2022/01/Medico-Clinic-Surgery-1024x681.jpg"
              />
            </div>
            <div
              class="carousel-item carousel-item"
            >
              <img
                alt="travel vacation"
                class="d-block w-100"
                src="https://www.carepointmedical.com.au/wp-content/uploads/2020/08/doctors_three.jpg"
              />
            </div>
          </div>
          <a
            class="carousel-control-prev"
            href="#"
            role="button"
            tabindex="0"
          >
            <span
              aria-hidden="true"
              class="carousel-control-prev-icon"
            />
            <span
              class="visually-hidden"
            >
              Previous
            </span>
          </a>
          <a
            class="carousel-control-next"
            href="#"
            role="button"
            tabindex="0"
          >
            <span
              aria-hidden="true"
              class="carousel-control-next-icon"
            />
            <span
              class="visually-hidden"
            >
              Next
            </span>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container-fluid"
    >
      <div
        class="login-card col-4 pb-5 bg-white border rounded container"
      >
        <div
          class="logo-wrapper"
        >
          <img
            alt="logo"
            class="logo-img"
            height="200"
            src="LOGO-DOCTOR.png"
            width="200"
          />
        </div>
        <form
          class=""
        >
          <div
            class="my-3"
          >
            <input
              class="form-control"
              id="formBasicEmail"
              placeholder="Email address"
              required=""
              type="email"
            />
          </div>
          <div
            class="mb-3 text-center"
          >
            <button
              class="btn btn-success"
              type="submit"
            >
              Login
            </button>
          </div>
        </form>
        <div>
          <p>
            Don't have an account?
             
            <span>
              <button
                class="access-btn text-primary"
                type="button"
              >
                Register
              </button>
            </span>
          </p>
        </div>
      </div>
      <div
        class="carousel carousel slide"
      >
        <div
          class="carousel-indicators"
        >
          <button
            aria-current="true"
            aria-label="Slide 1"
            class="active"
            data-bs-target=""
            type="button"
          />
          <button
            aria-current="false"
            aria-label="Slide 2"
            data-bs-target=""
            type="button"
          />
          <button
            aria-current="false"
            aria-label="Slide 3"
            data-bs-target=""
            type="button"
          />
          <button
            aria-current="false"
            aria-label="Slide 4"
            data-bs-target=""
            type="button"
          />
          <button
            aria-current="false"
            aria-label="Slide 5"
            data-bs-target=""
            type="button"
          />
          <button
            aria-current="false"
            aria-label="Slide 6"
            data-bs-target=""
            type="button"
          />
          <button
            aria-current="false"
            aria-label="Slide 7"
            data-bs-target=""
            type="button"
          />
        </div>
        <div
          class="carousel-inner"
        >
          <div
            class="carousel-item active carousel-item"
          >
            <img
              alt="travel vacation"
              class="d-block w-100"
              src="https://images.unsplash.com/photo-1629909613654-28e377c37b09?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxzZWFyY2h8Mnx8Y2xpbmljfGVufDB8fDB8fA%3D%3D&w=1000&q=80"
            />
          </div>
          <div
            class="carousel-item carousel-item"
          >
            <img
              alt="travel vacation"
              class="d-block w-100"
              src="https://e-clinic.co.uk/wordpress_new/wp-content/uploads/2021/09/Image-14-09-2021-at-13.59.jpg"
            />
          </div>
          <div
            class="carousel-item carousel-item"
          >
            <img
              alt="travel vacation"
              class="d-block w-100"
              src="https://firstmedcenters.com/wp-content/uploads/2022/07/English-speaking-clinic-FirstMed-V2-2.jpg"
            />
          </div>
          <div
            class="carousel-item carousel-item"
          >
            <img
              alt="travel vacation"
              class="d-block w-100"
              src="https://www.extraspace.com/blog/wp-content/uploads/2020/10/preparing-home-for-airbnb.jpg"
            />
          </div>
          <div
            class="carousel-item carousel-item"
          >
            <img
              alt="travel vacation"
              class="d-block w-100"
              src="https://img1.wsimg.com/isteam/stock/4810/:/cr=t:0%25,l:5.83%25,w:88.34%25,h:100%25"
            />
          </div>
          <div
            class="carousel-item carousel-item"
          >
            <img
              alt="travel vacation"
              class="d-block w-100"
              src="https://healthwaymedical.com/wp-content/uploads/2022/01/Medico-Clinic-Surgery-1024x681.jpg"
            />
          </div>
          <div
            class="carousel-item carousel-item"
          >
            <img
              alt="travel vacation"
              class="d-block w-100"
              src="https://www.carepointmedical.com.au/wp-content/uploads/2020/08/doctors_three.jpg"
            />
          </div>
        </div>
        <a
          class="carousel-control-prev"
          href="#"
          role="button"
          tabindex="0"
        >
          <span
            aria-hidden="true"
            class="carousel-control-prev-icon"
          />
          <span
            class="visually-hidden"
          >
            Previous
          </span>
        </a>
        <a
          class="carousel-control-next"
          href="#"
          role="button"
          tabindex="0"
        >
          <span
            aria-hidden="true"
            class="carousel-control-next-icon"
          />
          <span
            class="visually-hidden"
          >
            Next
          </span>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
